### PR TITLE
Fix swss::exec return value

### PR DESCRIPTION
**Pull request by qiluo-msft**
Fix https://github.com/Azure/sonic-swss-common/issues/327

### CODE REVIEW

**Comment by qiluo-msft** [comment]
Retest this please

**Comment by qiluo-msft** [comment]
@stepanblyschak Could you also help review?

File: common/exec.h [hunk 1]
```diff
@@ -4,6 +4,15 @@
 
 namespace swss {
 
+enum EXEC_RC
```
Comment (pavel-shirshov) [inline common/exec.h:7]: is it possible to use enum class here?

Comment (qiluo-msft) [reply common/exec.h:7]: Technically enum class is working but not convenient. I just need named constant integer, so enum is good enough.

---
In reply to: [460169126](https://github.com/Azure/sonic-swss-common/pull/368#discussion_r460169126) [](ancestors = 460169126)

Comment (pavel-shirshov) [reply common/exec.h:7]: enum class is c++ feature which helps gives us more type checking in compile time. Why is it non-convenient?

Comment (qiluo-msft) [reply common/exec.h:7]: There are no implicit conversions from the values of a scoped enumerator to integral types.

---
In reply to: [460172825](https://github.com/Azure/sonic-swss-common/pull/368#discussion_r460172825) [](ancestors = 460172825,460169126)

File: common/exec.cpp [hunk 2]
```diff
@@ -33,12 +33,43 @@ int exec(const string &cmd, string &stdout)
         }
     }
 
-    int ret = pclose(pipe);
-    if (ret != 0)
+    int ret;
+    string msg;
+    int status = pclose(pipe);
+    if (status == -1)
```
Comment (tahmed-dev) [inline common/exec.cpp:39]: nit: the chain of if else if.. lends itself well to a switch-case statement.

Comment (tahmed-dev) [reply common/exec.cpp:39]: nevermind. this will assume knowledge about the ranges each class of error code occupy.

File: common/exec.h (hunk 1 above)
Comment (pavel-shirshov) [reply common/exec.h:7]: Sorry. I didn't get. The feature of enum class I meant is you can compare C enum value with any integer value, and C++ enum class must be compared with this enum class instance only, any integer will not work. sonic-swss-common is C++, so should we use C++ features, not C?

Comment (qiluo-msft) [reply common/exec.h:7]: I agree `enum class` has improvement on `enum`, however, the type safety does not help/hurt my use case. I don't even use comparison.

I use them for readable `return NNN;` statements. Using `enum class` requires explicit type conversion, which hurts the readability and gains not much.

---
In reply to: [460183560](https://github.com/Azure/sonic-swss-common/pull/368#discussion_r460183560) [](ancestors = 460183560,460172825,460169126)

**Pull request merged**